ec/src/lib.rs

```diff
@@ -133,14 +102,15 @@ pub trait CurveGroup:
     + core::iter::Sum<Self::Affine>
     + for<'a> core::iter::Sum<&'a Self::Affine>
 {
-    type Config: CurveConfig<ScalarField = Self::ScalarField, BaseField = Self::BaseField>;
+    type Config: CurveConfig<ScalarField = <Self as PrimeGroup>::ScalarField, BaseField = Self::BaseField>;
+    type ScalarField: PrimeField;
```

Comment:
If `PrimeGroup` already has a `ScalarField` we don't need to repeat it here.

Reply:
We need to disambiguate between `PrimeGroup::ScalarField` and `Group::ScalarField` because rustc is stupid and doesn't understand they are the same.

Reply:
My question is more that why do we need to repeat `ScalarField` between `Group`, `PrimeGroup` and `CurveGroup`? Why not just specify it once in `Group` and be done?

Reply:
moved to `Scalar`